### ISSUE TITLE
Blogger images defaulting to https

### ISSUE THREAD

**Issue by Lozula** [OpenLiveWriter/OpenLiveWriter#227]
In WLW, when uploading images to my blogger blog, they would be hosted on picasa, and the image url would be something like:

http://lh3.googleusercontent.com/-UdGCWWyYkPg/VnlMzi6QPGI/AAAAAAAMBlU/KpdE4QQuBqY/moulin%252520rouge%252520Paris%252520experience_thumb%25255B1%25255D.jpg?imgmax=800

Now all the urls are https, so:

https://lh3.googleusercontent.com/-UdGCWWyYkPg/VnlMzi6QPGI/AAAAAAAMBlU/KpdE4QQuBqY/moulin%252520rouge%252520Paris%252520experience_thumb%25255B1%25255D.jpg?imgmax=800

This doesn't change anything in terms of the image as they work over both http and https, but I don't think it needs to be https as it takes longer to set up the secure connection, particularly with a post with lots of images.

**Comment by martinwoodward** [comment]
Do you have some data on what the performance impact is from using HTTPS for images on your blog?  I just did some tests from my connection loading your image from the HTTP and HTTPS URL's 100 times each using curl and I couldn't see any difference in the average download time (the HTTPS connection was actually marginally faster but in the run I did but was so close it was in the margin of error).

This was just for one image though and not a good real world test of the impact of multiple images on a page. I also don't have any caching proxy servers between me and Google that might improve the performance for cached (non-https) content.

**Comment by Lozula** [comment]
I ran some tests on tools.pingdom.com for a post where I switched between https and http on all the image on the post, and the load time per image goes from 20ms to 70ms between http and https when I flip between the two.

I'm not sure exactly how accurate that is of course, and since they load in parallel to the rest of the post it's probably not a huge impact on overall load time.

I also happily hold my hands up and don't claim to be any kind of expert on web page rendering or networking or how important this stuff is. I just noticed that this had changed between WLW and OLW and wasn't sure if it was intentional or had the potential to impact load time :)

This is the tools.pingdom overview of one image between http and https. 

![image https](https://cloud.githubusercontent.com/assets/16286517/11974872/8c8dd116-a968-11e5-8f5b-6cc6dc02faae.jpg)
![image non https](https://cloud.githubusercontent.com/assets/16286517/11974873/8c9774f0-a968-11e5-9001-1b246bd55fd3.jpg)

**Comment by martinwoodward** [comment]
Sounds good. If anyone is having a look at this, the code for getting the image URL's after they have been uploaded is here:

https://github.com/OpenLiveWriter/OpenLiveWriter/blob/master/src/managed/OpenLiveWriter.BlogClient/Clients/GoogleBloggerv3Client.cs

**Comment by dorianmuthig** [comment]
I was just browsing the issues, but when I saw this I just had to say something.
@Lozula: You deserve a hard knock on the head. Using TLS is good for you and your users. Then it takes 50ms longer to load a picture, so what? Getting a picture in under 100ms is quite the feat in itself, most of the web doesn't load that fast, and if you consider the internet speed of the average user, it in fact can't. The traffic overhead for TLS negotiation is minimal, the delay likely doesn't come from the negotiation step at all, but the computing resources required for encrypting what is to be sent down the wire. This means, the impact is negligible and you will be doing your visitors a favor, especially those who connect to open Wi-Fi networks.
I would propose closing this as **won't ever fix**. The only alternative I would see, is to always remove the protocol information from any URLs in `href` or `src` attributes, so that they'll look like `//subdomain.domain.tld/path/to/file.jpg`. That way, users in China, India and other regions where encryption is illegal can visit your website via plain HTTP and everyone else can enjoy TLS security and HSTS by visiting your website via HTTPS at least once.

**Comment by nvella** [comment]
Closing this issue as wontfix, as since Picasa was shut down and the Blogger client was switched to Drive for photo upload (#810), there is no possible way to retrieve content from Drive without an automatic redirect to HTTPS. We may be able to serve Blogger images over HTTP in the future through alternative image host support.
